### ISSUE TITLE
Different treatment for authors names in taxa versus sources

### ISSUE THREAD

**Issue by adrik29** [SpeciesFileGroup/taxonworks#1506]
Prepositions are used as part of person's names in many languages.
In Portuguese, we have the prepositions da, das, do, dos and de [= from; of].
The current convention in Portuguese is that they usually **do not make part** of a composite name, unless people artificially merge it into the name, as my friend Márcio Bernardino da Silva, who, unhappy by having the commonest surname in Brazil just created DaSilva (which, in my opinion is horrible, he could have gone with the middle name which is nowhere near as common).

Now, in TW: 

1) taxon names DO NOT USE PREPS.
2) source names USE PREPS.

This create anomalous things, such as:

Paratamboicus bicornutus Mello-Leitão, 1940 in de Mello-Leitão, 1940c: 97.
https://sfg.taxonworks.org/tasks/nomenclature/browse?taxon_name_id=323529

As if it were two different people.

**Comment by mjy** [comment]
In Taxon Names the verbatim name field can always be used to over-ride how a name looks in a Taxon Name authorship, use it when the person does not format correctly, if you want.

In general, however, we do not care about representing how things look, but rather the meaning behind them.  Resolving to the correct author (Person) is more important than representing the past.  The latter is already done in the paper, you can go back and see that there, why duplicate it when we can do better (a rhetorical question).

There is definitely an existing problem with properly capturing Portugese names, however I doubt we will be able to resolve it unless we can find a standard that defines what to do.  I.e. we are not developing our own standard.  Currently we use BibTeX approach to maximize utility provided by the software tools that surround its implementation.  Deviating from this is whack-a-mole, one issue gets better, 2 get worse.

**Comment by proceps** [comment]
@mjy I noticed this to, the problem come from the slight differences how the Author string is generated for the Source and for the TaxonName. For the Source the Author string is generated with prefixes, for the TaxonName, the prefix is omitted. In both cases the string is generated from people. The same people are used in both, the source and taxon_name. We have to try to unify this.  I can probably take a look at the issue...

**Comment by mjy** [comment]
I fear it will not be easy, since one method uses the bibtex renderer, the other not.  Changing to use the same method (bibtex, not custom) means that all taxon names will need to be re-indexed if/when we choose a default CLS style other than Zootaxa.

If you do look at it then start with writing specs.

**Comment by proceps** [comment]
@adrik29 What is actually the preferred way of citing the people who has prefixes in their last name. Giving your example. "de Mello-Leitão" vs. "Mello-Leitão".

So, in sources, it is much harder to fix. For the sources, we use a BibTeX parser, which suppose to standardize the names according to the roles applied by different journals. So I noticed, depending which prefix you use it either appear or not as a part of last name. In TaxonName author string we used different approach, and we literally used only the last name, and if prefix is used as part of last name and not divided from it, it will be parsed into a the name string. But all prefixes put in the prefix field, are specifically excluded.

So if you say, that in your example "de" should be displayed as part of compound last name, I can relatively easy add this and make the TaxonName author string to match the source Author string. If they match, you will not see the repeats in the Author string. 

If you say, that the Source author string is incorrect and "de" should not be included there. That is would mean, that I practically cannot do anything. Because this is the way BibTex parses the names.
Please let me know what do you think.

**Comment by adrik29** [comment]
@proceps : I wish to use what is standard for main world languages, not creating my own standard.

For Portuguese (Wikipedia): The particle 'de'
Different from in Italian surnames, these conjunctives are usually not part of a composite name, i.e., "Sousa" is not different from "de Sousa," and both are ordered under 'S' in an alphabetical list. Therefore, one should not refer to Luiz Pereira da Silva as Mr. Da Silva but rather Mr. Silva.

For Italian, on the contrary, they use the particle as part of the name.

Summing up,
Brazilian Cândido de Mello-Leitão should be listed under M = M-L, C. de
Italian Lodovico di Caporiacco should be listed under D = Di Caporiacco, L.

**Comment by adrik29** [comment]
I just wanted to have the same standards for the sources and for the authors, but as from I understand from Matt's comments it is not an easy accomplishment. We just can't have crazy stuff such as: "M-L" in "DE M-L".

Of course we have to be flexible to accommodate for each other preferences.
I, for one, hate stupid formats that many journals implement, such as:

(1) to ignore the fascicle numbers of periodicals (e.g., in ZJLS), when they sometimes are critical to locating the article;
(2) to use comma instead of colon before page numbers as Zootaxa does -- this is stupid: the whole world uses colon. And furthermore, Zootaxa is incongruent, because they use commas in the references, but the official citation of each article on the top of title page of theirs always has COLONS. Now this is a big overlook....

![image](https://user-images.githubusercontent.com/46284943/82680471-ae945800-9c22-11ea-814a-859746687cc0.png)

![image](https://user-images.githubusercontent.com/46284943/82680502-bbb14700-9c22-11ea-80b3-13810f2da9c2.png)



And as a last note, how can I get TW to show page numbers of books? sometimes is does, sometimes it doesn't, but I haven't discovered the trick...

**Comment by proceps** [comment]
I know Zootaxa format is frustrating for me too. I checked several other BibTeX output for other journals, ZooKeys for example, all of them invent something new. May be to avoid some proprietary formats. At least for now, I try to match the author string with the Source.Author string, if both use people and people match. I pushed changes, we have to wait for new migration.
